searx/webapp.py

```diff
@@ -354,6 +354,16 @@ def render(template_name, override_theme=None, **kwargs):
 
 @app.before_request
 def pre_request():
+    # request.request_data
+    if request.method == 'POST':
+        request_data = request.form
+    elif request.method == 'GET':
+        request_data = request.args
+    else:
+        request_data = {}
+
+    request.request_data = request_data
```

Comment:
Yes, using `request.form` is the good practice from that point, it contains all the available arguments after GET/POST merge.